yt/frontends/gadget/io.py

```diff
@@ -71,8 +71,9 @@ def _read_particle_coords(self, chunks, ptf):
                 y = f["/%s/Coordinates" % ptype][si:ei, 1].astype("float64")
                 z = f["/%s/Coordinates" % ptype][si:ei, 2].astype("float64")
                 if ptype == self.ds._sph_ptype:
-                    hsml = f[
-                        "/%s/SmoothingLength" % ptype][si:ei].astype("float64")
+                    pdtype = f["/%s/Coordinates" % ptype].dtype
+                    pshape = f["/%s/Coordinates" % ptype].shape
+                    hsml = self._get_smoothing_length(data_file, pdtype, pshape)
```

Comment:
This does change behavior, for Gadget data, right?  (Even if it's an improvement, I want to make sure I understand.)

Reply:
@MatthewTurk it shouldn't change anything, although I just noticed that I missed out the `astype("float64")` bit. look below in the call to `_get_smoothing_length`.

Reply:
:+1: